system/js/search.js

```diff
@@ -1,79 +1,118 @@
-	function dec_sort(a, b){
-		return ($(b).text()) > ($(a).text()) ? 1 : -1;    
-	}
-	//remove duplicate
-	function removeDup(a){
-		var seen = {};
-		$(a).each(function() {
-			var txt = $(this).text();
-				if (seen[txt])
-					$(this).remove();
-				else
-					seen[txt] = true;
-		});
-	}
+  //first condition
+  function firstCond(){
+    $('input').val("");
+    $('#listSearch .ui-listview-item').hide();
+  };
+  function unselect(){
+    $('#listSearch').find('.selected').removeClass('selected');
+    $('#listSearch').find('.active').removeClass('active');
+  }
+  // accending sort
+  function asc_sort(a, b){
+    return ($(b).text()) < ($(a).text()) ? 1 : -1;    
+  }
+  // decending sort
+  function dec_sort(a, b){
+    return ($(b).text()) > ($(a).text()) ? 1 : -1;    
+  }
+  //remove duplicate
+  function removeDup(a){
+    var seen = {};
+    $(a).each(function() {
+      var txt = $(this).text();
+        if (seen[txt])
+          $(this).remove();
+        else
+          seen[txt] = true;
+    });
+  }
+  function isUrlValid(url) {
+    var urlregex = /(ftp|http|https):\/\/(\w+:{0,1}\w*@)?(\S+)(:[0-9]+)?(\/|\/([\w#!:.?+=&%@!\-\/]))?/
+    return urlregex.test(url);
+  }
+
+  function visit_link(a){
+    Utils.run_command("x-www-browser "+a);
+  }
+
+  function run_terminal(a){
+    Utils.run_command('gnome-terminal -e "'+a+'"');
+  }
+  
 	var typingTimeout;
 	function listFilter(header, list) { 
-	// header is any element, list is an unordered list
-   // create and add the filter form to the header
+    // header is any element, list is an unordered list
+    // create and add the filter form to the header
     var form = $("<div>").attr({"class":"filterform"}),
         input = $("<input>").attr({"class":"filterinput","type":"text","id":"searchinput","placeholder":"Type to search..."});
+        runin = $("<div>").attr({"class":"runin"});
     $(form).append(input).appendTo(header);
+    $(form).append(runin).appendTo(header);
    
-	//arrow navigation
-   
-
+  
      $(input)
 		.click( function(){firstCond();unselect();})
+    //keyboard
 		.keyup( function (e) {
-			 var $item = $('#listSearch a span:visible').not(':has(:empty)');
+      var $item = $('#listSearch a span:visible').not(':has(:empty)');
 			var o = {38: 'up',40: 'bottom',37: 'prev',39: 'next'}
 			var dir = o[e.which];
-		    var $active = $('.active'),
-		        i = $item.index($active);
-			
-			
+      var $active = $('.active'),
+      i = $item.index($active);
+
+      //run applications
+      var filter = $(this).val();
 			if (e.which == 13) {
-		        //$(list).find('.selected').removeClass('selected');
-		        $('span.active').parent().addClass('selected');
-		        var a = $active.parent().attr('desktop')
-				Utils.run_desktop(a);
-				//alert(a);
-				firstCond();
-				unselect();
-		    }
-		    else if(e.which == 38 || e.which == 40 ){
+          //reload manokwari
+        if(filter.indexOf("!r")>=0 && filter.length==2){
```

Comment:
Use `===` instead of `==`

https://appendto.com/2016/02/vs-javascript-abstract-vs-strict-equality/